homeassistant/components/intellifire/fan.py

```diff
@@ -0,0 +1,142 @@
+"""Fan definition for Intellifire."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import math
+from typing import Any
+
+from intellifire4py import IntellifireControlAsync, IntellifirePollData
+
+from homeassistant.components.fan import (
+    FanEntity,
+    FanEntityDescription,
+    FanEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import ordered_list_item_to_percentage
+
+from .const import DOMAIN, LOGGER
+from .coordinator import IntellifireDataUpdateCoordinator
+from .entity import IntellifireEntity
+
+
+@dataclass
+class IntellifireFanRequiredKeysMixin:
+    """Required keys for fan entity."""
+
+    set_fn: Callable[[IntellifireControlAsync, int], Awaitable]
+    value_fn: Callable[[IntellifirePollData], bool]
+    data_field: str
+    named_speeds: list[str]
+
+
+@dataclass
+class IntellifireFanEntityDescription(
+    FanEntityDescription, IntellifireFanRequiredKeysMixin
+):
+    """Describes a fan entity."""
+
+
+INTELLIFIRE_FANS: tuple[IntellifireFanEntityDescription, ...] = (
+    IntellifireFanEntityDescription(
+        key="fan",
+        name="Fan",
+        set_fn=lambda control_api, speed: control_api.set_fan_speed(speed=speed),
+        value_fn=lambda data: data.fanspeed,
+        data_field="fanspeed",
+        named_speeds=[
+            "quiet",
+            "low",
+            "medium",
+            "high",
+        ],  # off is not included
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the fans."""
+    coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if coordinator.data.has_fan:
+        async_add_entities(
+            IntellifireFan(coordinator=coordinator, description=description)
+            for description in INTELLIFIRE_FANS
+        )
+        return
+    LOGGER.info("Disabling Fan - IntelliFire device does not appear to have one")
+
+
+class IntellifireFan(IntellifireEntity, FanEntity):
+    """This is Fan entity for the fireplace."""
+
+    entity_description: IntellifireFanEntityDescription
+    _attr_supported_features = FanEntityFeature.SET_SPEED | FanEntityFeature.PRESET_MODE
+
+    @property
+    def is_on(self):
```

Comment:
Could you add a type hint here?

Reply:
I'm surprised mypy didn't catch it... will do.

Reply:
mypy only checks existing type hints - not missing type hints unless you add intellifire to strict typing file

Reply:
Ah! ... time to add myself to strict typing i guess :)

Thanks